### ISSUE TITLE
[1.4.2.3] meteor npm install doesn't work with github on Windows

### ISSUE THREAD

**Issue by nathantreid** [meteor/meteor#8310]
I'm using Meteor 1.4.2.3 on Windows 10 x64, and I'm unable to meteor npm install any github packages, for example `meteor npm install github:vue-bulma/expanding`:

For testing, I installed npm 3.10.9 and node 4.6.2, and the `npm install` command works fine.

What I expect to see is this (this is the output when I run npm without meteor):
```
D:\projects\personal\test>npm i vue-bulma/expanding
D:\projects\personal\test
+-- UNMET PEER DEPENDENCY bulma@^0.2
+-- UNMET PEER DEPENDENCY vue@>=2.0.0-beta.0
`-- vue-bulma-expanding@0.0.0  (git://github.com/vue-bulma/expanding.git#c2699ed6cceb54563ab15c8b8fc87a7e5b65a855)

```

What I actually see is this:
```
D:\projects\personal\test>meteor npm i vue-bulma/expanding
npm ERR! git clone C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\git-github-com-vue-bulma-expanding-git-180e918b C:\Users\NATHAN~1\AppData\Local\Temp\npm-14468-c99eebc2\git-cache-e69424c0\c2699ed6cceb54563ab15c8b8fc87a7e5b65a855: Cloning into 'C:\Users\NATHAN~1\AppData\Local\Temp\npm-14468-c99eebc2\git-cache-e69424c0\c2699ed6cceb54563ab15c8b8fc87a7e5b65a855'...
npm ERR! git clone C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\git-github-com-vue-bulma-expanding-git-180e918b C:\Users\NATHAN~1\AppData\Local\Temp\npm-14468-c99eebc2\git-cache-e69424c0\c2699ed6cceb54563ab15c8b8fc87a7e5b65a855: error: cannot spawn sh: No such file or directory
npm ERR! git clone C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\git-github-com-vue-bulma-expanding-git-180e918b C:\Users\NATHAN~1\AppData\Local\Temp\npm-14468-c99eebc2\git-cache-e69424c0\c2699ed6cceb54563ab15c8b8fc87a7e5b65a855: fatal: unable to fork
npm ERR! git clone C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\git-github-com-vue-bulma-expanding-git-180e918b C:\Users\NATHAN~1\AppData\Local\Temp\npm-14468-c99eebc2\git-cache-e69424c0\c2699ed6cceb54563ab15c8b8fc87a7e5b65a855:
npm ERR! git clone C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\git-https-github-com-vue-bulma-expanding-git-76042935 C:\Users\NATHAN~1\AppData\Local\Temp\npm-14468-c99eebc2\git-cache-d45222e3\c2699ed6cceb54563ab15c8b8fc87a7e5b65a855: Cloning into 'C:\Users\NATHAN~1\AppData\Local\Temp\npm-14468-c99eebc2\git-cache-d45222e3\c2699ed6cceb54563ab15c8b8fc87a7e5b65a855'...
npm ERR! git clone C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\git-https-github-com-vue-bulma-expanding-git-76042935 C:\Users\NATHAN~1\AppData\Local\Temp\npm-14468-c99eebc2\git-cache-d45222e3\c2699ed6cceb54563ab15c8b8fc87a7e5b65a855: error: cannot spawn sh: No such file or directory
npm ERR! git clone C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\git-https-github-com-vue-bulma-expanding-git-76042935 C:\Users\NATHAN~1\AppData\Local\Temp\npm-14468-c99eebc2\git-cache-d45222e3\c2699ed6cceb54563ab15c8b8fc87a7e5b65a855: fatal: unable to fork
npm ERR! git clone C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\git-https-github-com-vue-bulma-expanding-git-76042935 C:\Users\NATHAN~1\AppData\Local\Temp\npm-14468-c99eebc2\git-cache-d45222e3\c2699ed6cceb54563ab15c8b8fc87a7e5b65a855:
npm ERR! git clone --template=C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\_templates --mirror git@github.com:vue-bulma/expanding.git C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\git-github-com-vue-bulma-expanding-git-9fcbcb55: Cloning into bare repository 'C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\git-github-com-vue-bulma-expanding-git-9fcbcb55'...
npm ERR! git clone --template=C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\_templates --mirror git@github.com:vue-bulma/expanding.git C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\git-github-com-vue-bulma-expanding-git-9fcbcb55: error: cannot spawn ssh: No such file or directory
npm ERR! git clone --template=C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\_templates --mirror git@github.com:vue-bulma/expanding.git C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\git-github-com-vue-bulma-expanding-git-9fcbcb55: fatal: unable to fork
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Users\\nathantreid\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.4.2_3\\mt-os.windows.x86_32\\dev_bundle\\bin\\node.exe" "C:\\Users\\nathantreid\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.4.2_3\\mt-os.windows.x86_32\\dev_bundle\\lib\\node_modules\\npm\\bin\\npm-cli.js" "i" "vue-bulma/expanding"
npm ERR! node v4.6.2
npm ERR! npm  v3.10.9
npm ERR! code 128

npm ERR! Command failed: git -c core.longpaths=true clone --template=C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\_templates --mirror git@github.com:vue-bulma/expanding.git C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\git-github-com-vue-bulma-expanding-git-9fcbcb55
npm ERR! Cloning into bare repository 'C:\Users\nathantreid\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\dev_bundle\.npm\_git-remotes\git-github-com-vue-bulma-expanding-git-9fcbcb55'...
npm ERR! error: cannot spawn ssh: No such file or directory
npm ERR! fatal: unable to fork
npm ERR!
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     D:\projects\personal\test\npm-debug.log
```

As soon as I experienced this I switched to another PC running the same version of Windows and Meteor, and got the same result.

**Comment by benjamn** [comment]
Do you have an `ssh.exe` executable installed (and in your path)?

**Comment by nathantreid** [comment]
No - I installed OpenSSH and now it works (thanks!), but why did `npm` work when `meteor npm` didn't?

**Comment by gcacars** [comment]
```
D:\Dropbox\SITES\Sistemas\app>meteor update
C:\Users\gabri\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\isopackets\ddp\npm\node_modules\meteor\promise\node_modules\meteor-promise\promise_server.js:190
      throw error;
      ^

Error: spawn EPERM
    at exports._errnoException (util.js:911:11)
    at ChildProcess.spawn (internal/child_process.js:309:11)
    at Object.exports.spawn (child_process.js:362:9)
    at C:\tools\cli\main.js:527:44
    at springboard (C:\tools\cli\main.js:525:18)
    at C:\tools\cli\main.js:1501:7
 => awaited here:
    at Promise.await (C:\Users\gabri\AppData\Local\.meteor\packages\meteor-tool\1.4.2_3\mt-os.windows.x86_32\isopackets\ddp\npm\node_modules\meteor\promise\node_modules\meteor-promise\promise_server.js:39:12)
    at springboard (C:\tools\cli\main.js:531:8)
    at C:\tools\cli\main.js:1501:7
```

**Comment by hwillson** [comment]
Glad to hear adding OpenSSH helped fixed the issue!  have a feeling the lack of OpenSSH wasn't the real issue though. As you mentioned, `npm install github:vue-bulma/expanding` was working properly without it. Looking at the error message you got back, it shows 3 separate attempts to clone the repo when using `meteor npm install`, which lines up with npm's [git clone attempt cycle](https://github.com/npm/npm/blob/d46015256941ddfff1463338e3e2f8f77624a1ff/lib/cache/add-remote-git.js#L78-L110). It tries `git` first, followed by `git+https`, followed by `git+ssh`, which lines up with your errors. The fact that the first `git` attempt should have worked without needing ssh but failed for some reason, is puzzling.

It sounds like there could be a bug here somewhere, but given that this hasn't been reported that frequently, and current versions of Meteor are running with npm 4 (which includes many bug fixes), I think we're safe to close this issue for now. Happy to re-open if this comes up again. Thanks!

@gcacars - I'm not sure what your comment relates to. It doesn't seem related to this issue.

**Comment by mikicho** [comment]
I think I'm having same issue here's [log file](https://gist.github.com/mikicho/38d453201c2b8f89ed99bbbbfd8e84e6)

**Comment by abernix** [comment]
@mikicho What happens if you run?:

    git clone git@github.com:reactioncommerce/transliteration.git

**Comment by mikicho** [comment]
@abernix Sorry for the late response, I was broad, the command succeeded:
```
remote: Counting objects: 1377, done.
remote: Total 1377 (delta 0), reused 0 (delta 0), pack-reused 1377
Receiving objects: 100% (1377/1377), 1.16 MiB | 580.00 KiB/s, done.
Resolving deltas: 100% (623/623), don
```

**Comment by bgadrian** [comment]
I was getting a different error, I've lost a few hours because it didn't said what is missing actually (ssh). 
I'm using GitBash so I didn't had OpenSSH (that solved this issue), on Windows 10.
I've found the solution on a meteor project (not related) https://github.com/reactioncommerce/reaction/issues/2939

`git-upload-pack: command not foundnpm ERR! fatal: Could not read from remote repository.`

When using npm-install it worked (to install remote dependencies), but I have problems in the web-browser JS (the module was not in the bundle), so I tried to use `meteor npm install`, which lead me to this problem.
